author: Sune Vuorela <debian@pusling.com>

fixes compilation issues of programs using qbuttons.h with gcc4.2

--- a/src/widgets/qbutton.h
+++ b/src/widgets/qbutton.h
@@ -183,7 +183,7 @@ inline const QPixmap *QButton::pixmap() 
 
 inline bool QButton::isToggleButton() const
 {
-    return toggleTyp != SingleShot;
+    return ToggleType(toggleTyp) != SingleShot;
 }
 
 inline  bool QButton::isDown() const
@@ -193,7 +193,7 @@ inline  bool QButton::isDown() const
 
 inline bool QButton::isOn() const
 {
-    return stat != Off;
+    return ToggleState(stat) != Off;
 }
 
 #ifndef QT_NO_COMPAT
